shared/react-native/android/app/src/main/java/io/keybase/ossifrage/modules/ShareFiles.java

```diff
@@ -31,11 +36,28 @@ public String getName() {
     @ReactMethod
     public void share(String uriPath, String mimeType, Promise promise) {
         File file = new File(uriPath);
-        Uri fileUri = FileProvider.getUriForFile(reactContext, "io.keybase.ossifrage.fileprovider", file);
-
-        Intent intent = new Intent(Intent.ACTION_SEND)
-                .putExtra(Intent.EXTRA_STREAM, fileUri)
-                .setType(mimeType);
+        Intent intent = new Intent(Intent.ACTION_SEND).setType(mimeType);
+        if (mimeType.contains("text/")) {
+            try {
+                BufferedReader br = new BufferedReader(new FileReader(file));
+                StringBuilder textBuilder = new StringBuilder();
+                String text;
+                for (int i=0; i < MAX_TEXT_FILE_LINES && (text = br.readLine()) != null; i++) {
```

Comment:
Does this mean we can't share text files that are larger than 1000 lines?

Reply:
We can, it'll only share the first 1000 lines though. We'll see if `text/*` works.

Reply:
No bueno on `text/*`